Use Standard Dependency Workflow

joshuatcasey opened the pull request:
## Summary
Use the standard dependency workflow. Resolves #289 

Note that the logic in `dependency/actions/compile/entrypoint` seems much simpler than the logic in [`buildpacks-ci`](https://github.com/cloudfoundry/buildpacks-ci/blob/f4cf2db0338254e5c346b7c22d34fe8fc0338cdb/tasks/build-binary-new/builder.rb#L31-L62), yet it seems to achieve the same result. This is verified by running the compilation step for `22.2.2` and comparing the result to https://deps.paketo.io/pip/pip_22.2.2_linux_noarch_bionic_97bb6b2d.tgz.

Note the version of the included `setuptools`. In `buildpacks-ci` this seems to be pinned to `62.1.0`, yet in https://deps.paketo.io/pip/pip_22.2.2_linux_noarch_bionic_97bb6b2d.tgz it's clearly `63.4.1`. Why is that? Can we allow it to just float to the latest?

robdimsdale commented on .github/.syncignore:
```diff
@@ -1,2 +1,2 @@
 CODEOWNERS
-workflows/update-dependencies-from-metadata.yml
+workflows/update-dependencies.yml
```
If we delete the `update-dependencies.yml` workflow file, should we also remove this entry from the `syncignore`?

joshuatcasey replied:
We should leave it in `.github/.syncignore` so that the `github-config` sync doesn't re-add the file.

robdimsdale reviewed (approved):
This looks great, with one small question about the `syncignore` file.

robdimsdale commented:
Thanks for putting this together @joshuatcasey .

I think this simplified logic for `pip` download/install is appropriate. In the past we've had to do more complex things (like pinning `setuptools` to a specific version) due to incompatibilities between versions of `pip` and `python`. We may find we have to do that again, should certain versions of `pip` not work on certain versions of `python`.

joshuatcasey commented:
@robdimsdale Could you reapprove? I rearranged the commits and rebased on main.

I also removed `linux_64` from the artifact name. The artifact produced is just reorganized source files.
